lib/audit.js

```diff
@@ -25,6 +31,63 @@ auditCmd.completion = function (opts, cb) {
   }
 }
 
+class Auditor extends Installer {
+  constructor (where, dryrun, args, opts) {
+    super(where, dryrun, args, opts)
+    this._deepArgs = (opts && opts.deepArgs) || []
+    this._runId = opts.runId || ''
+    this.audit = false
+  }
+
+  loadAllDepsIntoIdealTree (cb) {
+    BB.fromNode(cb => super.loadAllDepsIntoIdealTree(cb)).then(() => {
+      if (this._deepArgs && this._deepArgs.length) {
+        this._deepArgs.forEach(arg => {
+          arg.reduce((acc, child, i) => {
+            if (!acc) {
+              // We might not always be able to find `target` through the given
+              // path. If we can't we'll just ignore it.
+              return
+            }
+            const spec = npa(child)
+            const target = (
+              acc.requires.find(n => n.package.name === spec.name) ||
+              acc.requires.find(
+                n => audit.scrub(n.package.name, this._runId) === spec.name
```

Comment:
Long term we'll be undoing scrubbing in `submitForFullReport`, but this is fine till then, and conveniently compatible.

Reply:
that'll be nice!